unittests/src/main/java/uk/ac/manchester/tornado/unittests/vectortypes/TestDoubles.java

```diff
@@ -215,4 +215,107 @@ public void testDotProductDouble() {
 
         assertEquals(seqReduce[0], outputReduce[0], 0.001);
     }
+
+    public static void addVectorDouble2(VectorDouble2 a, VectorDouble2 b, VectorDouble2 results) {
+        for (@Parallel int i = 0; i < a.getLength(); i++) {
+            results.set(i, Double2.add(a.get(i), b.get(i)));
+        }
+    }
+
+    @Test
+    public void testVectorDouble2() {
+        int size = 256;
+
+        VectorDouble2 a = new VectorDouble2(size);
+        VectorDouble2 b = new VectorDouble2(size);
+        VectorDouble2 output = new VectorDouble2(size);
+
+        for (int i = 0; i < size; i++) {
+            a.set(i, new Double2(i, i));
+            b.set(i, new Double2(size - i, size - i));
+        }
+
+        //@formatter:off
+        new TaskSchedule("s0")
+                .task("t0", TestDoubles::addVectorDouble2, a, b, output)
```

Comment:
It seems that the Vector unit-tests test only additions. I think we should expand to cover all the supported vector operations for all types.